pkg/azdiskschedulerextender/azdiskschedulerextender.go

```diff
@@ -125,45 +128,59 @@ func filter(context context.Context, schedulerExtenderArgs schedulerapi.Extender
 	}()
 
 	requestedVolumes := schedulerExtenderArgs.Pod.Spec.Volumes
-	klog.V(2).Infof("Pod %s has requested %d volume/s.", schedulerExtenderArgs.Pod.Name, len(requestedVolumes))
+	klog.V(2).Infof("pod %s has requested %d volume/s", schedulerExtenderArgs.Pod.Name, len(requestedVolumes))
 
-	// if no volumes are requested, return assigning 0 score to all nodes
-	if len(requestedVolumes) != 0 {
-		nodesChan := make(chan azDriverNodesMeta)
-		// all available cluster nodes
-		allNodes := schedulerExtenderArgs.Nodes.Items
+	// if no volumes are requested, don't filter the nodes
+	if len(requestedVolumes) == 0 {
+		klog.V(2).Infof("no volumes are requested by pod %s, no further filtering is required", schedulerExtenderArgs.Pod.Name, len(requestedVolumes))
+		return formatFilterResult(schedulerExtenderArgs.Nodes.Items, schedulerExtenderArgs.NodeNames, failedNodes, ""), nil
+	}
 
-		go getAzDriverNodes(context, nodesChan)
-		azDriverNodesMeta := <-nodesChan
-		if azDriverNodesMeta.err != nil {
-			filteredNodes = schedulerExtenderArgs.Nodes.Items
-			filteredNodeNames = schedulerExtenderArgs.NodeNames
-			klog.Warningf("Failed to get the list of azDriverNodes: %v", azDriverNodesMeta.err)
+	nodesChan := make(chan azDriverNodesMeta)
+	// all node candidates
+	allNodes := schedulerExtenderArgs.Nodes.Items
 
-			return formatFilterResult(filteredNodes, filteredNodeNames, failedNodes, ""), nil
-		}
+	go getAzDriverNodes(context, nodesChan)
+	azDriverNodesMeta := <-nodesChan
+	if azDriverNodesMeta.err != nil {
+		klog.Warningf("failed to get the list of azDriverNodes: %v. Further filtering is skipped", azDriverNodesMeta.err)
+		return formatFilterResult(schedulerExtenderArgs.Nodes.Items, schedulerExtenderArgs.NodeNames, failedNodes, ""), nil
+	}
 
-		// map node name to azDriverNode state
-		nodeNameToStatusMap := make(map[string]bool)
-		for _, azDriverNode := range azDriverNodesMeta.nodes {
-			if azDriverNode.Status != nil {
-				nodeNameToStatusMap[azDriverNode.Spec.NodeName] = *azDriverNode.Status.ReadyForVolumeAllocation
-			}
+	// map node name to azDriverNode state
+	nodeNameToAzDriverNodeStatusMap := make(map[string]bool)
+	for _, azDriverNode := range azDriverNodesMeta.nodes {
+		if azDriverNode.Status != nil {
+			nodeNameToAzDriverNodeStatusMap[azDriverNode.Spec.NodeName] = *azDriverNode.Status.ReadyForVolumeAllocation
```

Comment:
Since `ReadyForVolumeAllocation` is a pointer, we should defend against a `nil` value here in addition to `azDriverNode.Status` being `nil`.